Update to ghcide-0.3.0

jkachmar opened the pull request:
This PR updates the `niv` fetcher expression to the latest version at the time of writing (v23) and also updates `ghcide` to track the `v0.3.0` release branch (which was recently pushed out).

I haven't tested this locally yet, and I haven't really looked much into the CI process for this repo so if there are any additional manual steps let me know and I'll take a look.

jkachmar commented on nix/sources.json:
```diff
@@ -1,14 +1,14 @@
         "repo": "ghcide",
-        "rev": "c8a982cb1aff4e1a91b03aaa8cb60a6adbdc5eb7",
-        "sha256": "0hsb5zyadpv2rjzaghwqcj8lngfa8mq6rb05lk16zzsb4v8bz85l",
+        "rev": "tags/v0.3.0",
```
To semi-significant changes here with respect to the old version:
1. `ghcide` is now under the [haskell](https://www.github.com/haskell) organization
    - the 0.3 release seems to track the latest Stackage nightly, which is on GHC 8.10.2
    - hopefully this means that the [cachix/ghcide](https://github.com/cachix/ghcide) fork shouldn't be necessary anymore
2. `niv` points to the `v0.3.0` tag rather than `master` or a particular branch
    - `niv add haskell/ghcide --rev tags/v0.3.0` worked for this
    - `niv update ghcide --rev tags/vX.Y.Z` should work to update for future tagged releases

jkachmar replied:
Ah, well I hadn't noticed that https://github.com/haskell/ghcide/pull/751 didn't make the v0.3.0 cutoff; oh well I've reverted back to the latest commit on `ghcide`'s main branch since that has the up-to-date GHC 8.10.2 `stack.yaml` file.

jkachmar commented:
Getting this error in CI; I'll take a look later today (EDT) to try and resolve:

```
error: The Haskell package set does not contain the package: stm (build dependency).
```

jkachmar commented:
It looks like [`haskell.nix`](github.com/input-output-hk/haskell.nix) has changed since this project was last updated; there are problems with the Hercules CI build and it looks like they're suggesting people use IOHK's Hydra instance for binary caching.

cf. https://github.com/input-output-hk/haskell.nix/pull/668

jkachmar commented:
Just confirmed locally that using the IOHK Hydra cache (as outlined [here](https://iohk.zendesk.com/hc/en-us/articles/900000673963-Installing-Nix-and-setting-up-IOHK-binaries)) avoids the caching problem.

jkachmar commented:
It seems like GHC 8.4.4 isn't necessarily available in IOHK's cache anymore; I'm not getting spurious GHC builds if I avoid the GHC 8.4.4 derivation.

This commit may be related: https://github.com/input-output-hk/haskell.nix/commit/099d830db177c828e9fab919446dd7fd8d160dbb

domenkozar commented:
It seems fine to drop 8.4.4, those using it can use an old commit.

Setting up IOHK binary cache makes sense to speed up the builds.

jkachmar commented:
I've dropped GHC 8.4.4 and updated the action to pull from IOHK's Hydra cache.

Let me know if you'd like me to re-add the IOHK cachix cache; I dropped it because I didn't think it would be terribly useful, but maybe that would give us better odds of a cache hit on some of these dependencies.

Presumably after this is run agains the main branch, the cachix action will drop build times back down to 20 minutes or so (like it currently is).

domenkozar commented:
Thanks!
